docs/src/apps/jsonrpc-api.md

```diff
@@ -395,12 +395,35 @@ The signatures will be ordered based on the Slot in which they were confirmed in
 
 ```bash
 // Request
-curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedSignaturesForAddress","params":["6H94zdiaYfRfPfKjYLjyr2VFBg6JHXygy84r3qhc3NsC", 0, 100]}' localhost:8899
+curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedSignaturesForAddressDepricated","params":["6H94zdiaYfRfPfKjYLjyr2VFBg6JHXygy84r3qhc3NsC", 0, 100]}' localhost:8899
 
 // Result
 {"jsonrpc":"2.0","result":{["35YGay1Lwjwgxe9zaH6APSHbt9gYQUCtBWTNL3aVwVGn9xTFw2fgds7qK5AL29mP63A9j3rh8KpN1TgSR62XCaby","4bJdGN8Tt2kLWZ3Fa1dpwPSEkXWWTSszPSf1rRVsCwNjxbbUdwTeiWtmi8soA26YmwnKD4aAxNp8ci1Gjpdv4gsr","4LQ14a7BYY27578Uj8LPCaVhSdJGLn9DJqnUJHpy95FMqdKf9acAhUhecPQNjNUy6VoNFUbvwYkPociFSf87cWbG"]},"id":1}
 ```
 
+### getConfirmedSignaturesForAddress
+
+Returns confirmed signatures for transactions involving an
+address backwards in time from the provided slot
+
+#### Parameters:
+* `<string>` - account address as base-58 encoded string
+* `<object>` - (optional) Configuration object containing the following field:
+  * `startAfter: <string>` - start searching from this transaction signature,
+                             which must be a confirmed signature for the account
+                             address.  If not provided the search starts from
+                             the highest max confirmed block.
```

Comment:
I'm a little confused by the different `start`s here. So, if a signature is provided, we search from that signature forward for 1000 transactions, but if no signature is provided, we search from the highest max confirmed block backwards for 1000 transactions?

Reply:
It always search backwards, I'll clarify

Reply:
Ah, cool. Should the response be ordered from highest to lowest slot, then?

Reply:
yep, fixed the doc

Reply:
Awesome, lgtm. Do you want me to do the implementation?

Reply:
Way to take my 🍬 away.   Sure, that'd be great actually.  Then I can jump into the storage bucket stuff with the assumption that it can just be plugged in easily.

For a mainnet-beta migration, I was thinking:
1. First update the mainnet-beta/devnet/testnet RPC nodes to a version that includes `deprecatedGetConfirmedSignaturesForAddress` and `getConfirmedSignaturesForAddress`, backed by the same implementation.   Rename `getConfirmedSignaturesForAddress` to `deprecatedGetConfirmedSignaturesForAddress` in the documentation too. 
2. Now update web3.js to use `deprecatedGetConfirmedSignaturesForAddress`, and employ that new version in the `explorer`
3. Replace `getConfirmedSignaturesForAddress` with the new interface/impl and update mainnet-beta/devnet/testnet RPC nodes`.  Add new `getConfirmedSignaturesForAddress` to the docs, and remove`deprecatedGetConfirmedSignaturesForAddress`.
4. Adapt the cli to the new `getConfirmedSignaturesForAddress` in `process_transaction_history()` and update web3.js to use `getConfirmedSignaturesForAddress`, and employ that new version in the `explorer`.
5. Delete the `deprecatedGetConfirmedSignaturesForAddress` implementation on the `master` branch only.  It can ship with 1.3, so there's a window for any external users (if any) to upgrade before it disappears